File: src/XStat.Service/Models/SystemInfo.cs
namespace XStat.Service.Models;

/// <summary>
/// Static hardware/system info that doesn't change at runtime (CPU model, RAM
/// size/speed, disk drive letters, OS version). Exposed via GET /api/system-info
/// so the frontend can show labels like "i7-12700K", "32GB", "Windows 11 23H2".
/// </summary>
public record SystemInfo(
    string OsName,           // "Windows 11 Pro"
    string OsVersion,        // "23H2 (Build 22631.4317)"
    string CpuModel,         // "12th Gen Intel Core i7-12700K"
    string GpuModel,         // "NVIDIA GeForce RTX 3080"
    long   RamTotalBytes,    // total physical RAM in bytes
    uint   RamSpeedMhz,      // memory clock speed (MT/s), 0 if unknown
    string RamType,          // "DDR4" / "DDR5" / ""
    IReadOnlyList<RamStick> RamSticks,
    IReadOnlyList<DiskInfo>  Disks
);

public record RamStick(
    string Device,           //Locator/Slot: "DIMM0" etc.
    string Manufacturer,
    long   CapacityBytes,
    uint   SpeedMhz,
    string PartNumber,
    string FormFactor        // "SODIMM" / "DIMM" / ""
);

public record DiskInfo(
    string Model,            // "Samsung SSD 980 PRO 500GB"
    string DriveLetter,      // "C:" (may be empty for unmounted)
    string Label,            // "本地磁盘" / "System"
    long   TotalBytes,
    long   FreeBytes,
    string Type              // "SSD" / "HDD" / ""
);

File: src/app/src/types/systemInfo.ts
// Static system info matching the .NET backend SystemInfo model.
// Fetched rarely (info doesn't change at runtime) via GET /api/systeminfo.

export interface RamStick {
  device: string
  manufacturer: string
  capacityBytes: number
  speedMhz: number
  partNumber: string
  formFactor: string
}

export interface DiskInfo {
  model: string
  driveLetter: string
  label: string
  totalBytes: number
  freeBytes: number
  type: string
}

export interface SystemInfo {
  osName: string
  osVersion: string
  cpuModel: string
  gpuModel: string
  ramTotalBytes: number
  ramSpeedMhz: number
  ramType: string
  ramSticks: RamStick[]
  disks: DiskInfo[]
  uptimeSeconds?: number
}
